src/scitokens_internal.h

```diff
@@ -802,6 +922,36 @@ class Validator {
         }
     }
 
+    /**
+     * Helper method to record monitoring statistics for validation errors.
+     */
+    void record_validation_error(const std::string &issuer,
+                                  const std::exception &e,
+                                  const std::chrono::steady_clock::time_point &start_time) {
+        auto end_time = std::chrono::steady_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(
+            end_time - start_time);
+        
+        std::string error_msg = e.what();
+        
+        // Check if this is a failed issuer lookup (network/DNS errors)
```

Comment:
I don't like introspecting strings.  Instead, add an exception hierarchy that helps with this.